pkg/gameserversets/gameserversets.go

```diff
@@ -75,12 +78,45 @@ func sortGameServersByLeastFullNodes(list []*agonesv1.GameServer, count map[stri
 				return false
 			}
 		}
-		// if both Nodes have the same count, one node is emptied first (packed scheduling behavior)
+
+		// Check Node total count
 		acTotal, bcTotal := ac.Allocated+ac.Ready, bc.Allocated+bc.Ready
-		if acTotal == bcTotal {
-			return a.Status.NodeName < b.Status.NodeName
+		if acTotal < bcTotal {
+			return true
 		}
-		return acTotal < bcTotal
+		if acTotal > bcTotal {
+			return false
+		}
+
+		// if the Nodes have the same count and CountsAndLists flag is enabled, sort based on CountsAndLists Priorities.
+		if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) && (priorities != nil) {
+			for _, priority := range priorities {
+				res := compareGameServerCapacity(&priority, a, b)
+				switch priority.Order {
+				case agonesv1.GameServerPriorityAscending, "":
```

Comment:
I would say at this point, we can assume there is a default, and if there isn't, that's a bug.